### ISSUE TITLE
Infinite Scroll on home page and all pages with projects

### ISSUE THREAD

**Issue by GriffGreen** [Giveth/giveth-dapps-v2#1906]
**As a lazy user, I want to keep scrolling and just see new projects pop up.**

**Comment by laurenluz** [comment]
@mateodaza - what ever happened to this feautre?

**Comment by laurenluz** [comment]
It was a feature we had a loooong time ago - and it was cool! When we did the full upgrade to the newer design early this year (or maybe when we upgraded to typescript?) I remember we got rid of it and I thought it was a functionality thing.

It definitely makes for better UX. It's a feature I miss having all the time... and @GriffGreen just mentioned it the other day too.

If there's no objectionsf from @markoprljic or @mosaeedi I think we should do it.

**Comment by markoprljic** [comment]
On the Homepage there can be Load more button (up to a reasonable amount of items loaded, to avoid going heavy on resources), and the See all projects button to allow user to go to the Projects page. No infinite scroll on the Homepage because user must be able to see the Footer and other information preceding it. 

On the Projects page (where there is not much in and before the Footer) we **can** have infinite scroll. 

Typically how it's done these days is that there is a Load more button (in case script gets stuck or bad connection on client side), when user reaches the end of the first list the auto-load has a really short delay before it loads a new batch of items. This way we assume that the user want's to see more projects and automatically load it for them, but in case user scrolls over that section really fast, the auto load will not trigger and user can see the footer.

**Comment by laurenluz** [comment]
Thanks @markoprljic - sounds good to me!

@RamRamez I would say this is not high priority, but a nice to have when we can.

cc @MoeNick

**Comment by markoprljic** [comment]
I'm in favor of Mo's suggestion, even better.

**Comment by MoeNick** [comment]
@RamRamez can you jump into this or you need some changes from backend. as Mo said we have to load 45 projects after 1st click.

**Comment by RamRamez** [comment]
20 projects will be shown on first request, after scrolling down there will be infinite scroll and lazy loading of projects (loading 20 more projects each time), until 45 projects are loaded. So on the third request only 5 projects will be loaded then `load more` button will be shown.

@mosaeedi Is there a chance to show 60 projects on infinite scroll instead of 45?

**Comment by MoeNick** [comment]
I guess you talked @RamRamez @mosaeedi and it's clear for you Ramin? Can we make it happen?

**Comment by RamRamez** [comment]
> I guess you talked @RamRamez @mosaeedi and it's clear for you Ramin? Can we make it happen?

Yes, I'm on it.

**Comment by maryjaf** [comment]
I tested on staging, 15 project are shown on first step, after scrolling down there will be infinite scroll and lazy loading of projects, loading 15 more projects each time until 45 projects are loaded and then the **load more** button is shown
Is this the expected result? @RamRamez @MoeNick

**Comment by RamRamez** [comment]
> I tested on staging, 15 project are shown on first step, after scrolling down there will be infinite scroll and lazy loading of projects, loading 15 more projects each time until 45 projects are loaded and then the **load more** button is shown Is this the expected result? @RamRamez @MoeNick

Yes, it's ok ([link](https://github.com/Giveth/giveth-dapps-v2/issues/1906#issuecomment-1349062474))

**Comment by GriffGreen** [comment]
Exciting!
